appcues/src/main/java/com/appcues/ui/composables/AppcuesComposition.kt

```diff
@@ -126,37 +126,24 @@ private fun BoxScope.ComposeLastRenderingState(state: Rendering) {
             // apply backdrop traits
             ApplyBackgroundDecoratingTraits(backdropDecoratingTraits.value)
 
-            // TODO contentPadding + stepDecoratingPadding into one? to fix tooltip padding issue
             // create wrapper
-            contentWrappingTrait.WrapContent { hasFixedHeight, contentPadding ->
+            contentWrappingTrait.WrapContent { modifier, wrapperInsets ->
                 Box(contentAlignment = Alignment.TopCenter) {
-                    ApplyUnderlayContainerTraits(containerDecoratingTraits.value)
+                    ApplyUnderlayContainerTraits(containerDecoratingTraits.value, wrapperInsets)
 
                     // Apply content holder trait
                     with(contentHolderTrait) {
                         // create object that will passed down to CreateContentHolder
                         ContainerPages(
                             pageCount = steps.size,
                             currentPage = state.position,
-                            composePage = { index ->
-                                with(steps[index]) {
-                                    CompositionLocalProvider(
-                                        LocalAppcuesActions provides actions,
-                                        LocalExperienceStepFormStateDelegate provides formState
-                                    ) {
-                                        // used to get the padding values from step decorating trait and apply to the Column
-                                        val density = LocalDensity.current
-                                        val stepDecoratingPadding = remember(this) { StepDecoratingPadding(density) }
-
-                                        ApplyUnderlayStepTraits(this@Box, stepDecoratingPadding)
-
-                                        ComposeStepContent(index, hasFixedHeight, contentPadding, stepDecoratingPadding)
-
-                                        ApplyOverlayStepTraits(this@Box, stepDecoratingPadding)
-
-                                        ComposeStickyContent(this@Box, stepDecoratingPadding)
-                                    }
-                                }
+                            composePage = {
+                                steps[it]
+                                    .ComposeStep(
+                                        modifier = modifier.testTag("page_$it"),
```

Comment:
since we have the modifier parameter we can set the testTag here instead of passing the index to do it later.